lua/luasnip/session/snippet_collection.lua

```diff
@@ -270,6 +284,10 @@ function M.add_snippets(snippets, opts)
 			table.insert(by_prio[snip.snippetType][snip.priority][ft], snip)
 			table.insert(by_ft[snip.snippetType][ft], snip)
 			by_id[snip.id] = snip
+
+			if opts.source and opts.store_meta_data then
+				snip["meta_data"] = snippets_meta_data[#snippets_meta_data]
```

Comment:
One downside of storing a direct reference to that table in the snippet is that it would be copied recursively, which I don't think we want. I feel like an id per add-snippets-call and associating the snippets with that id would be the right choice, but I can't really rationalize

Reply:
Aaah so we just add a table in the middle which maps all ids to the respective meta table and then get meta data by `id_to_meta[snip.id]`

Reply:
I think using `snip.id` might be the easiest way to implement it for now, but yes, essentially that's what I'm getting at :D

Reply:
I think this is what i did in my first few commits with a getter fn that gets the meta data by snip id

Reply:
Nice, I think that that is the best choice for us.
Giving a snippet another id is also a bit superfluous since we can just map id -> other_id